jwql/jwql_monitors/generate_preview_images.py

```diff
@@ -694,13 +701,13 @@ def process_program(program):
             # insert it and it's associated DQ array into the
             # instance of PreviewImage. Also set the input
             # filename to indicate that we have mosaicked data
-            if numfiles != 1:
-                im.data = mosaic_image
-                im.dq = mosaic_dq
-                im.file = dummy_file
+            # if numfiles != 1:
```

Comment:
Is this commented out in order to get around the error seen in development, or do you mean to comment this out permanently?

Reply:
Ah yes, sorry.  I had commented that out to get around the mosaic issue described above.  Commenting this out made the code work, but the issue should be addressed instead of just leaving this commented.  Here is a log file that contains the full error:

`<central_storage_area>/logs/dev/generate_preview_images/generate_preview_images_2021-05-04-10-18.log`

Reply:
@bhilbert4 I am just checking in on this -- were you able to determine what the issue was here?  If not, no worries.  But I am wondering if we should just merge this PR in and make a separate ticket for the mosaic issue.  Let me know what you think. Thanks!

Reply:
I have a couple guesses: 
1. There are two if clauses that look at the length of `file_list`. The first looks for a length > 1 in order to create the mosaic. The other looks for a length != 1 in order to assign the mosaic to the data attribute. So if the file_list happened to be an empty list, with length 0, then the mosaic would not be created, but it would try to populate the data model with it. 
2. The traceback shows a problem within multiprocessing. We've been having a lot of (unresolved) issues with multiprocessing in Mirage, but only for python >=3.8, which is not the case here.

So if you want to do a quick test, you could change the if clause that contains `im.data = mosaic_image` to look for a length > 1 (rather than != 1) and see if that helps. If we really do have a case of an empty file list though, then I wonder if there would be problems downstream.

Reply:
Cool, thanks for the ideas @bhilbert4, I will try these out!

Reply:
Per your suggestions and changes in https://github.com/spacetelescope/jwql/pull/726/commits/b29b4dbb4fdf00ccdada63c2fd137414ecc71307, I was able to fix the bug with the mosaics.